File: flutter_app/lib/screens/splash_screen.dart
import 'package:flutter/material.dart';
import '../theme/app_theme.dart';
import '../widgets/glass.dart';
import '../services/auth_service.dart';
import 'login_screen.dart';
import 'dashboard_shell.dart';

/// Cinematic 3-second brand intro, then routes to login or dashboard.
class SplashScreen extends StatefulWidget {
  const SplashScreen({super.key});

  @override
  State<SplashScreen> createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen>
    with SingleTickerProviderStateMixin {
  late final AnimationController _c =
      AnimationController(vsync: this, duration: const Duration(milliseconds: 2600))..forward();

  @override
  void initState() {
    super.initState();
    _go();
  }

  Future<void> _go() async {
    await Future.delayed(const Duration(seconds: 3));
    if (!mounted) return;
    final loggedIn = AuthService().isLoggedIn;
    Navigator.of(context).pushReplacement(
      PageRouteBuilder(
        transitionDuration: const Duration(milliseconds: 550),
        pageBuilder: (_, __, ___) =>
            loggedIn ? const DashboardShell() : const LoginScreen(),
        transitionsBuilder: (_, a, __, child) =>
            FadeTransition(opacity: a, child: child),
      ),
    );
  }

  @override
  void dispose() {
    _c.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final logoIn = CurvedAnimation(parent: _c, curve: const Interval(0.0, 0.6, curve: Curves.easeOutCubic));
    final textIn = CurvedAnimation(parent: _c, curve: const Interval(0.45, 1.0, curve: Curves.easeOut));

    return Scaffold(
      body: GlassBackground(
        child: Center(
          child: Column(
            mainAxisSize: MainAxisSize.min,
            children: [
              AnimatedBuilder(
                animation: _c,
                builder: (context, _) {
                  return Transform.scale(
                    scale: 0.6 + 0.4 * logoIn.value,
                    child: Opacity(
                      opacity: logoIn.value.clamp(0.0, 1.0),
                      child: Container(
                        width: 116,
                        height: 116,
                        decoration: BoxDecoration(
                          color: const Color(0xFFF2F6FC),
                          borderRadius: BorderRadius.circular(30),
                          boxShadow: NeuDecoration.raised(distance: 6, blur: 16, alpha: 0.5),
                        ),
                        child: ClipRRect(
                          borderRadius: BorderRadius.circular(30),
                          child: Image.asset('assets/images/logo.png',
                              fit: BoxFit.cover),
                        ),
                      ),
                    ),
                  );
                },
              ),
              const SizedBox(height: 26),
              FadeTransition(
                opacity: textIn,
                child: SlideTransition(
                  position: Tween(begin: const Offset(0, 0.4), end: Offset.zero)
                      .animate(textIn),
                  child: Column(
                    children: [
                      Text('EZ-Workspace',
                          style: TextStyle(
                            fontSize: 26,
                            fontWeight: FontWeight.w900,
                            letterSpacing: -0.5,
                            color: AppColors.ink,
                          )),
                      const SizedBox(height: 6),
                      Text('ENTERPRISE OPERATIONS PANEL',
                          style: TextStyle(
                            fontSize: 10,
                            fontWeight: FontWeight.w700,
                            letterSpacing: 3,
                            color: AppColors.muted,
                          )),
                    ],
                  ),
                ),
              ),
              const SizedBox(height: 44),
              FadeTransition(
                opacity: textIn,
                child: SizedBox(
                  width: 34,
                  height: 34,
                  child: CircularProgressIndicator(
                    strokeWidth: 2.6,
                    valueColor: AlwaysStoppedAnimation(AppColors.primary),
                    backgroundColor: AppColors.primary.withValues(alpha: 0.15),
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

File: android_native/app/src/main/java/com/ezbillify/ezworkspace/ui/screens/SplashScreen.kt
package com.ezbillify.ezworkspace.ui.screens

import androidx.compose.animation.core.Animatable
import androidx.compose.animation.core.EaseOutCubic
import androidx.compose.animation.core.tween
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ezbillify.ezworkspace.R
import com.ezbillify.ezworkspace.services.AuthService
import com.ezbillify.ezworkspace.theme.AppColors
import com.ezbillify.ezworkspace.theme.InterFontFamily
import com.ezbillify.ezworkspace.theme.neumorphicRaised
import kotlinx.coroutines.delay

/** Cinematic 3-second brand intro, then routes to login or dashboard. */
@Composable
fun SplashScreen(onDone: (loggedIn: Boolean) -> Unit) {
    val scale = remember { Animatable(0.6f) }

    LaunchedEffect(Unit) {
        scale.animateTo(1f, animationSpec = tween(650, easing = EaseOutCubic))
        delay(2350)
        onDone(AuthService().isLoggedIn)
    }

    Box(
        modifier = Modifier.fillMaxSize().background(AppColors.scaffold),
        contentAlignment = Alignment.Center,
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Box(
                modifier = Modifier
                    .size(116.dp)
                    .scale(scale.value)
                    .neumorphicRaised(cornerRadius = 30.dp),
            ) {
                Image(
                    painter = painterResource(R.drawable.logo),
                    contentDescription = "EZ-Workspace",
                    contentScale = ContentScale.Crop,
                    modifier = Modifier.fillMaxSize().clip(RoundedCornerShape(30.dp)),
                )
            }
            Spacer(Modifier.size(26.dp))
            Text(
                "EZ-Workspace",
                fontFamily = InterFontFamily, fontWeight = FontWeight.Black, fontSize = 26.sp,
                color = AppColors.ink,
            )
            Spacer(Modifier.size(6.dp))
            Text(
                "ENTERPRISE OPERATIONS PANEL",
                fontFamily = InterFontFamily, fontWeight = FontWeight.Bold, fontSize = 10.sp,
                color = AppColors.muted,
            )
            Spacer(Modifier.size(44.dp))
            CircularProgressIndicator(color = AppColors.primary, strokeWidth = 2.6.dp, modifier = Modifier.size(34.dp))
        }
    }
}
